File: sample-code/storage-app/aws-s3/presignedURLs.py
import boto3
import requests
import json

client_id = ''
token = ''

region = ''
bucket = ''
src_key = 'path/to/your/image.jpg'
dest_key = 'path/to/where/you/want/your/output.jpg'


def get_presigned_url(bucket, key, operation, expires_in, region):
    s3 = boto3.client('s3', region_name=region)
    url = s3.generate_presigned_url(operation, Params={'Bucket': bucket, 'Key': key}, ExpiresIn=expires_in)
    return url

input_scr_url = get_presigned_url(bucket, src_key, 'get_object', 3600, region)
output_dest_url = get_presigned_url(bucket, dest_key, 'put_object', 3600, region)

File: sample-code/storage-app/aws-s3/presignedURLs.js
/*************************************************
# install aws-sdk
npm install aws-sdk
# executescript
REGION=<AWS_REGION_YOU_BUCKET_IS_IN> BUCKET=<YOUR_BUCKET_NAME> OBJECT_KEY=<ANYTHING_YOU_WANT> OPERATION=<getObject | putObject> node presignedURLs.js
# example of GET pre-signed url
REGION=us-east-1 BUCKET=john-bucket OBJECT_KEY=some/path/to/image.psd OPERATION=getObject node presignedURLs.js
# example of PUT pre-signed url
REGION=us-east-1 BUCKET=john-bucket OBJECT_KEY=some/path/to/where/you/want/the/ouptut.jpg OPERATION=putObject node presignedURLs.js
************************************************/
var AWS = require('aws-sdk')
var bucket = process.env.BUCKET
var object_key = process.env.OBJECT_KEY
var operation =  process.env.OPERATION || 'putObject'
var region = process.env.REGION
AWS.config.update({"region": region});


let params = {Bucket: bucket, Key: object_key, Expires: 3600 * 24};
let s3 = new AWS.S3({apiVersion: '2006-03-01', signatureVersion: 'v4'});
s3.getSignedUrl(operation, params, function (err, url) {
  if(err){
    console.log(err)
  }else{
    console.log(url)
  }
});
